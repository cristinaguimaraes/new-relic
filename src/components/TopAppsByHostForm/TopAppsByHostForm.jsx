import React, { useState } from "react";
import {
  Helper,
  Input,
  Label,
  Form,
  Button,
  ButtonWrapper,
  Title
} from "./styled";
import { func } from "prop-types";

const TopAppsByHostForm = ({ onDeleteApp, onShowToastMessage, ...props }) => {
  const [host, setHost] = useState();
  const [error, setError] = useState();

  const handleOnSubmit = event => {
    event.preventDefault();

    if (!host) {
      setError("Please add the host");
    } else {
      props.history.push(`/TopAppsByHost/${host}`);
      setHost("");
    }
  };

  const handleOnChange = event => {
    setHost(event.target.value);
    setError("");
  };

  return (
    <Form onSubmit={handleOnSubmit}>
      <Title>Top Apps By Host</Title>
      <Label>Host:</Label>
      <Input name="name" type="text" value={host} onChange={handleOnChange} />
      {error && <Helper>{error}</Helper>}
      <ButtonWrapper>
        <Button type="submit">Show List</Button>
      </ButtonWrapper>
    </Form>
  );
};

TopAppsByHostForm.propTypes = {
  onDeleteApp: func.isRequired,
  onShowToastMessage: func.isRequired
};

export { TopAppsByHostForm };
