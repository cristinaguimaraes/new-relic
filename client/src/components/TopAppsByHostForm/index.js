import { TopAppsByHostFormContainer } from './TopAppsByHostFormContainer';
export { TopAppsByHostFormContainer as TopAppsByHostForm };