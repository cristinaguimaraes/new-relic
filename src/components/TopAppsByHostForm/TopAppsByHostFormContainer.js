import { TopAppsByHostForm } from "./TopAppsByHostForm";
import { onDeleteApp, onShowToastMessage } from "redux/actions/creators";
import { withRouter } from "react-router-dom";
import { connect } from "react-redux";

const mapDispatchToProps = {
  onDeleteApp,
  onShowToastMessage
};

export const TopAppsByHostFormContainer = withRouter(
  connect(null, mapDispatchToProps)(TopAppsByHostForm)
);
