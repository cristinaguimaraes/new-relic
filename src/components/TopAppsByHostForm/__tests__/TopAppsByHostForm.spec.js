import { TopAppsByHostForm } from "../TopAppsByHostForm";
import { shallow } from "enzyme";
import React from "react";

describe("TopAppsByHostForm component", () => {
  test("should render properly", () => {
    const fn = () => {};
    expect(
      shallow(
        <TopAppsByHostForm
          history={{ push: fn }}
          onShowToastMessage={fn}
          onDeleteApp={fn}
        />
      )
    ).toMatchSnapshot();
  });
});
