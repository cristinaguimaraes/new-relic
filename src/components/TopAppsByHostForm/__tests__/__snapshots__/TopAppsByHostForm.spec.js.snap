// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopAppsByHostForm component should render properly 1`] = `
<styled.form
  onSubmit={[Function]}
>
  <styled.div>
    Top Apps By Host
  </styled.div>
  <styled.label>
    Host:
  </styled.label>
  <styled.input
    name="name"
    onChange={[Function]}
    type="text"
    value=""
  />
  <styled.div>
    <styled.button
      type="submit"
    >
      Show List
    </styled.button>
  </styled.div>
</styled.form>
`;
